lib/functions.php

```diff
@@ -54,8 +54,8 @@ function resolve(string $basePath, string $newPath): string
             $path = $delta['path'];
         } else {
             // Removing last component from base path.
-            $path = $base['path'];
-            $length = strrpos((string) $path, '/');
+            $path = (string) $base['path'];
+            $length = strrpos($path, '/');
             if (false !== $length) {
                 $path = substr($path, 0, $length);
```

Comment:
with the detailed types, and level 9, phpstan found this. 
To be really safe, we cast `$path` to a string straight away. Then its use at line 60 - passing to `substr()` is also safe.